orc8r/gateway/python/magma/ctraced/trace_manager.py

```diff
@@ -0,0 +1,104 @@
+"""
+Copyright 2020 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import logging
+import os
+import pathlib
+import subprocess
+
+from subprocess import SubprocessError
+
+TRACE_FILE_NAME = "call_trace.pcap"
+
+class TraceManager:
+    """
+    TraceManager is a wrapper for tshark specifically for starting and stopping
+    call/interface/subscriber traces.
+
+    Only a single trace can be captured at a time.
+    """
+    def __init__(self, config):
+        self._is_active = False # is call trace being captured
+        self._proc = None
+        self._file_name = None
+        self._trace_directory = config.get("trace_directory",
+                                           "/var/opt/magma/trace")
+        self._trace_interface = config.get("trace_interface", "eth0")
+
+
+    def start_trace(self) -> bool:
+        """
+        Start a call trace.
+        Captures all packets across the eth0 interface.
+        TODO (andreilee): Make configurable for trace type
+        :return: True if successful
+        """
+        if self._is_active:
+            logging.error("Failed to start trace: Trace already active")
+            return False
+        command = ["tshark"]
```

Comment:
Can you move this command also to yml file, so that we can use tcpdump if necessary.

Reply:
just curious, but are the tshark and tcpdump arguments equivalent?

Reply:
Not equivalent. Moved to separate factory classes for generating `tshark` or `tcpdump` commands.